enterprise/server/backends/authdb/authdb.go

```diff
@@ -25,19 +29,108 @@ import (
 
 const (
 	apiKeyLength = 20
+
+	// For encrypted keys, the first N characters serve as the nonce.
+	apiKeyNonceLength = 6
+
+	// Encrypted API keys have this prefix in the database.
+	apiKeyEncryptedValuePrefix = "en$"
```

Comment:
Is it because clearing out the value would violate the uniqueness constraint? If so would it work to have the migration make both `value,encrypted_value` part of a unique composite index and then drop the unique index on `value`?

Reply:
Good idea.